features/graph/api/src/main/java/org/opennms/netmgt/graph/api/generic/GenericGraphContainer.java

```diff
@@ -138,4 +128,68 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(graphs, properties);
     }
+    
+    public static GenericGraphContainerBuilder builder() {
+        return new GenericGraphContainerBuilder();
+    }
+    
+    public static class GenericGraphContainerBuilder {
+
+        private final static Logger LOG = LoggerFactory.getLogger(GenericGraphContainerBuilder.class);
+
+        // allow graphs to be replaced in builder : use a Map
+        private final Map<String, GenericGraph> graphs = new HashMap<>();
+        private final Map<String, Object> properties = new HashMap<>();
+        
+        private GenericGraphContainerBuilder() {}
+        
+        public GenericGraphContainerBuilder id(String id) {
+            property(GenericProperties.ID, id);
+            return this;
+        }
+        
+        public GenericGraphContainerBuilder label(String label){
+            property(GenericProperties.LABEL, label);
+            return this;
+        }
+        
+        public GenericGraphContainerBuilder description(String description) {
+            property(GenericProperties.DESCRIPTION, description);
+            return this;
+        }
+
+        public GenericGraphContainerBuilder property(String name, Object value){
+            if(name == null || value == null) {
+                LOG.debug("Property name ({}) or value ({}) is null => ignoring it.", name, value);
+                return this;
+            }
+            properties.put(name, value);
+            return this;
+        }
+        
+        public GenericGraphContainerBuilder properties(Map<String, Object> properties){
+            Objects.requireNonNull(properties, "properties cannot be null");
+            for (Map.Entry<String, Object> entry : this.properties.entrySet()) {
+                property(entry.getKey(), entry.getValue());
+            }
+            return this;
+        }
+        
+        public GenericGraphContainerBuilder containerInfo(GraphContainerInfo containerInfo) {
```

Comment:
Can we make this `applyContainerInfo` or something else? 
We are not setting the containerInfo but use the data from it.
This is different from the other methods which actually set the data the method suggest they would.

Reply:
ok. Thats a bit of a grey area since we also copy some things (e.g. collections) in other methods. But I changed it.